fabric-events-interaction-v0/src/main/java/net/fabricmc/fabric/api/event/BlockBreakEffectsCallback.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+/**
+ * Callback for the effects displayed when a block is broken (particles and sounds)
```

Comment:
Would it be possible to detach the particles and sounds from each other? Since things seem to be done via `syncWorldEvent`, it's yet another case of magic values.

Reply:
MC handle both particles and sound as one world event, to separate them you would have to make this mixin client-only, and mixin to a messy switch statement.